adapter-tcp/src/main/java/io/nosqlbench/adapter/tcpclient/TcpClientAdapterSpace.java

```diff
@@ -0,0 +1,113 @@
+import java.net.Socket;
+
+public class TcpClientAdapterSpace {
+
+    private final static Logger logger = LogManager.getLogger(TcpServerAdapterSpace.class);
+    private final NBConfiguration config;
+    Writer writer;
+    private PrintWriter console;
+
+    public TcpClientAdapterSpace(NBConfiguration config) {
+        this.config = config;
+        this.writer = createPrintWriter();
+    }
+
+    protected PrintWriter createPrintWriter() {
+
+        SocketFactory socketFactory = SocketFactory.getDefault();
+        boolean sslEnabled = config.getOptional(boolean.class, "ssl").orElse(false);
+        if (sslEnabled) {
+            NBConfiguration sslCfg = SSLKsFactory.get().getConfigModel().extractConfig(config);
+            socketFactory = SSLKsFactory.get().createSocketFactory(sslCfg);
+        }
+
+        String host = config.getOptional("host").orElse("localhost");
+        int port = config.getOptional(int.class, "port").orElse(8080);
+
+        try {
+            Socket socket = socketFactory.createSocket(host, port);
+            logger.info("connected to " + socket.toString());
+            return new PrintWriter(socket.getOutputStream());
+        } catch (IOException e) {
+            throw new RuntimeException("Error opening socket:" + e, e);
+        }
+    }
+
+    public void writeflush(String text) {
+        try {
+            writer.write(text);
+            writer.flush();
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+    public static NBConfigModel getConfigModel() {
+        return ConfigModel.of(TcpClientAdapterSpace.class)
+            .add(SSLKsFactory.get().getConfigModel())
+            .add(
+                Param.defaultTo("host","localhost")
+                    .setDescription("")
+            )
+            .add(
+                Param.defaultTo("port",8080)
+                    .setDescription("")
+            )
+            .add(
+                Param.defaultTo("filename","tcpclient")
+                    .setDescription("this is the name of the output file. If 'stdout', output is sent to tcpclient, not a file.")
+            )
+            .add(
+                Param.defaultTo("newline",true)
+                    .setDescription("whether to automatically add a missing newline to the end of any output\n")
+            )
+            .add(
+                Param.optional("format")
+                    .setRegex("csv|readout|json|inlinejson|assignments|diag")
+                    .setDescription("Which format to use.\n" +
+                        "If provided, the format will override any statement formats provided by the YAML. " +
+                        "If 'diag' is used, a diagnostic readout will be provided for binding constructions.")
+            )
+            .add(
+                Param.defaultTo("bindings","doc")
+                    .setDescription("This is a simple way to specify a filter for the names of bindings that you want to use.\n" +
```

Comment:
One option here is to utilize the Java text block.  You can remove concatenation, quotes, etc. for the string and replace with the following:

"""
<body of text>
"""